Src/library/net/network/socket/network_socket.e

```diff
@@ -206,35 +238,73 @@ feature -- Status setting
 
 	set_timeout (n: INTEGER)
 			-- Set timeout to `n' seconds.
+		obsolete
+			"Use `set_timeout_ns (ns)` [2018-01-15]"
 		require
 			non_negative: n >= 0
 		do
-			if n > 0 then
-				timeout := n
-			else
-				timeout := default_timeout
-			end
+			set_timeout_ns (n.to_natural_64)
 		ensure
 			timeout_set: timeout = n or timeout = default_timeout
 		end
 
+	set_timeout_ns (a_timeout_nanoseconds: NATURAL_64)
+			-- Set timeout with to `a_timeout_nanoseconds` nanoseconds.
+			-- warning: the timeout granularity of the platform may not be nanoseconds, but micro or milliseconds.
+		require
+			positive_timeout: a_timeout_nanoseconds /= 0
+			is_valid_timeout_ns: is_valid_timeout_ns (a_timeout_nanoseconds)
+		do
+			if a_timeout_nanoseconds = 0 then
+				timeout_ns := one_second_in_nanoseconds * default_timeout.to_natural_64
+			else
+				timeout_ns := a_timeout_nanoseconds
+			end
+		end
+
 	set_recv_timeout (a_timeout_seconds: INTEGER)
 			-- Set the receive timeout in seconds on Current socket.
 			-- if `0' the related operations will never timeout.
+		obsolete
+			"Use `set_recv_timeout_ns` using nanoseconds [2018-01-15]"
 		require
 			positive_timeout: a_timeout_seconds >= 0
 		do
-			c_set_sock_recv_timeout (descriptor, level_sol_socket, a_timeout_seconds)
+			set_recv_timeout_ns (one_second_in_nanoseconds * a_timeout_seconds.to_natural_64)
+		end
+
+	set_recv_timeout_ns (a_timeout_nanoseconds: NATURAL_64)
+			-- Set the receive timeout with `a_timeout_nanoseconds` nanoseconds on Current socket.
+			-- if `0' the related operations will never timeout.
```

Comment:
Capitalization at the beginning of a sentence.

Reply:
fixed